frontend/pages/dataset/_id/annotation-mode/index.vue

```diff
@@ -5,7 +5,24 @@
         :key="datasetName && workspace"
         :datasetId="datasetId"
         :breadcrumbs="breadcrumbs"
+        :showTrainButton="true"
+        @on-click-train="showTrainModal(true)"
       />
+      <BaseModal
+        :modal-custom="true"
+        :prevent-body-scroll="true"
+        modal-class="modal-auto"
+        modal-position="modal-top-center"
+        :modal-visible="visibleTrainModal"
+        allow-close
+        @close-modal="showTrainModal(false)"
+      >
+        <DatasetTrainComponent
+          datasetTask="FeedbackTask"
+          :datasetName="datasetName"
+          :workspaceName="workspace"
+        />
+      </BaseModal>
```

Comment:
I think the logic to show or not the modal for dataset trainning should be encapsulate inside the `datasetTrain.component` and we could have an additional component for th datasetTrain modal called `datasetTrain.modal`.

This could be tackle in another PR of refacto => could you add two `// TODO `comments one for encapsulate modal component for datasetTrain and the other to remove all logic regarding the showModal from the page ?